### PR TITLE
Split PR part 6: Alter worker_pool to match new auth, change space support

### PR DESCRIPTION
**Pull request by stopthatastronaut**
Slowly bringing everything into line...

### CODE REVIEW

File: docs/octopus_deploy_worker_pool.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+```ruby
+describe octopus_deploy_worker_pool('My Worker Pool').in_space("My Space") do
+  it { should exist } 
+end
+```
+
+#### Type
+
+```ruby
```
Comment (matt-richardson) [inline docs/octopus_deploy_worker_pool.md]: should this have the link to the auth doc?

File: lib/octopus_serverspec_extensions/type/octopus_deploy_worker_pool.rb [hunk 2]
```diff
@@ -28,20 +34,44 @@ def initialize(serverUrl, apiKey, worker_pool_name)
       @worker_pool = get_worker_pool_via_api(serverUrl, apiKey, worker_pool_name)
     end
 
+    def in_space(space_name)
+      # allows us to tag .in_space() onto the end of the resource. as in
+      # describe octopus_account("account name").in_space("MyNewSpace") do
```
Comment (matt-richardson) [inline lib/octopus_serverspec_extensions/type/octopus_deploy_worker_pool.rb]:
```suggestion
      # describe octopus_worker_pool("my pool name").in_space("MyNewSpace") do
```

File: lib/octopus_serverspec_extensions/type/octopus_deploy_worker_pool.rb [hunk 3]
```diff
@@ -28,20 +34,44 @@ def initialize(serverUrl, apiKey, worker_pool_name)
...
+      @spaceId = get_space_id?(space_name)
+      if @worker_pool_name.nil?
+        raise "'worker_pool_name' was not provided. Unable to connect to Octopus server to validate configuration."
```
Comment (matt-richardson) [inline lib/octopus_serverspec_extensions/type/octopus_deploy_worker_pool.rb]: We've already connected to the server to get space id?

Comment (stopthatastronaut) [reply lib/octopus_serverspec_extensions/type/octopus_deploy_worker_pool.rb]: As the previous type, we only connect at the last opportunity now

File: docs/octopus_deploy_worker_pool.md (hunk 1 above)
Comment (stopthatastronaut) [reply docs/octopus_deploy_worker_pool.md]: Added

**Pull request merged**